### DISCUSSION TITLE
Atari 7800 Supergames bank switching not working?

### DISCUSSION THREAD

**Discussion by smesgr9000** [sanni/cartreader#895] [Need help]
Hi everyone,

I have a "hw 5" device and can't dump 7800 games with use supergame bank switching mode. I'm not sure if this is an soldering or timing issue or a software bug

What works:
- Mega Drive dump works, sadly have no SNES cartridge (so port could be flanky)
- C64 dump works, sadly does carts also don't use bank switching
- A7800 dump on "normal" roms without bank switching is good

issue:
- A7800 dump on supergames always read bank 7 over and over. Those games are good and dump with Atarimax without an issue

tried so far:
- comment out the bank writing code does not change ROM
- adding more NOPs between instructions does not change ROM
- changing writing address to from 0x8000 to 0xBFFF does not change ROM

measurements:
- adding keypress after each instruction, removed cartridge and probed the pins after each "breakpoint"
-> R/W High/Low is set according to Pin 1
-> Address pin A15 is set high, others low
-> Data pins are set to low

has anybody has success with those bank switching or any hints what to check next?

**Comment by PsyK0p4T** [comment]
Same behavior ? https://github.com/sanni/cartreader/discussions/878

**Comment by smesgr9000** [comment]
thank for the hint. This explains alot. I will investigate next weekend.
